pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -0,0 +1,126 @@
+    new_transaction.original_query_size = query_len;
+    bpf_map_update_elem(&postgres_in_flight, conn_tuple, &new_transaction, BPF_ANY);
+}
+
+// Handles a command complete message by enqueuing the transaction and deleting it from the in-flight map.
+static __always_inline void handle_command_complete(conn_tuple_t *conn_tuple, postgres_transaction_t *transaction) {
+    transaction->response_last_seen = bpf_ktime_get_ns();
+    postgres_batch_enqueue_wrapper(conn_tuple, transaction);
+    bpf_map_delete_elem(&postgres_in_flight, conn_tuple);
+}
+
+SEC("socket/postgres_process")
+int socket__postgres_process(struct __sk_buff* skb) {
+    skb_info_t skb_info = {};
+    conn_tuple_t conn_tuple = {};
+
+    if (!fetch_dispatching_arguments(&conn_tuple, &skb_info)) {
+        return 0;
+    }
+
+    normalize_tuple(&conn_tuple);
+
+    // Read first message header
+    struct pg_message_header header;
+    if (!read_message_header(skb, &skb_info, &header)) {
+        return 0;
+    }
+    // Advance the data offset to the end of the first message header.
+    skb_info.data_off += sizeof(struct pg_message_header);
+
+    // If the message is a new query, we store the query in the in-flight map.
+    // If we had a transaction for the connection, we override it and drops the previous one.
+    if (header.message_tag == POSTGRES_QUERY_MAGIC_BYTE) {
+        // message_len includes size of the payload, 4 bytes of the message length itself, but not the message tag.
+        // So if we want to know the size of the payload, we need to subtract the size of the message length.
+        handle_new_query(skb, &skb_info, &conn_tuple, header.message_len - sizeof(__u32));
+        return 0;
+    }
+
+    // We didn't find a new query, thus we assume we're in the middle of a transaction.
+    // We look up the transaction in the in-flight map, and if it doesn't exist, we ignore the message.
+    postgres_transaction_t *transaction = bpf_map_lookup_elem(&postgres_in_flight, &conn_tuple);
+    if (!transaction) {
+        return 0;
+    }
+
+    // If the message is a command complete, we enqueue the transaction and delete it from the in-flight map.
+    if (header.message_tag == POSTGRES_COMMAND_COMPLETE_MAGIC_BYTE) {
+        handle_command_complete(&conn_tuple, transaction);
+        return 0;
+    }
+
+    // We're in the middle of a transaction, and the message is not a command complete, but it can be a chain of
+    // messages. So we try to read up to POSTGRES_MAX_MESSAGES messages, looking for a command complete message.
+
+    // Advance the data offset to the end of the first message (after the payload). The message length includes the size
+    // of the payload, 4 bytes of the message length itself, but not the message tag. Since we already moved the data
+    // offset to the end of the message header, we want to jump over the payload.
+    skb_info.data_off += header.message_len - sizeof(__u32);
+
+#pragma unroll(POSTGRES_MAX_MESSAGES)
```

Comment:
Lets add to the documentation some links to the RFC, Here can be a nice place to link the documnation for the command complete message, and the option to response with multiple responses.

As well let's link the message format document - https://www.postgresql.org/docs/current/protocol-message-formats.html somewhere, maybe here before the handle_new_query function